src/main/java/seedu/address/logic/commands/RemoveCommand.java

```diff
@@ -294,5 +321,26 @@ public void setTags(Set<Tag> tags) {
         public Optional<Set<Tag>> getTags() {
             return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
         }
+
+        @Override
+        public boolean equals(Object other) {
+            // short circuit if same object
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof RemovePersonDescriptor)) {
+                return false;
+            }
+
+            // state check
+            RemovePersonDescriptor r = (RemovePersonDescriptor) other;
+
+            return getName().equals(r.getName())
+                    && getPhone().equals(r.getPhone())
+                    && getLessonPlan().equals(r.getLessonPlan())
+                    && getTags().equals(r.getTags());
```

Comment:
can remove all the unnecessary fields

Reply:
i meant to remove like phone, name, lessonplan entirely from removepersondescriptor not remove the whole equals method 😭😭 

just compare the relevant ones like homework grade attendance in equals method only